app/src/main/java/com/keylesspalace/tusky/FiltersActivity.kt

```diff
@@ -47,119 +47,100 @@ class FiltersActivity : BaseActivity() {
+                {
                     Toast.makeText(this@FiltersActivity, "Error creating filter '$phrase'", Toast.LENGTH_SHORT).show()
                 }
-            }
-
-            override fun onFailure(call: Call<Filter>, t: Throwable) {
-                Toast.makeText(this@FiltersActivity, "Error creating filter '$phrase'", Toast.LENGTH_SHORT).show()
-            }
-        })
-    }
-
-    private fun showAddFilterDialog() {
-        val binding = DialogFilterBinding.inflate(layoutInflater)
-        binding.phraseWholeWord.isChecked = true
-        AlertDialog.Builder(this@FiltersActivity)
-            .setTitle(R.string.filter_addition_dialog_title)
-            .setView(binding.root)
-            .setPositiveButton(android.R.string.ok) { _, _ ->
-                createFilter(binding.phraseEditText.text.toString(), binding.phraseWholeWord.isChecked)
-            }
-            .setNeutralButton(android.R.string.cancel, null)
-            .show()
-    }
-
-    private fun setupEditDialogForItem(itemIndex: Int) {
-        val binding = DialogFilterBinding.inflate(layoutInflater)
-        val filter = filters[itemIndex]
-        binding.phraseEditText.setText(filter.phrase)
-        binding.phraseWholeWord.isChecked = filter.wholeWord
-
-        AlertDialog.Builder(this@FiltersActivity)
-            .setTitle(R.string.filter_edit_dialog_title)
-            .setView(binding.root)
-            .setPositiveButton(R.string.filter_dialog_update_button) { _, _ ->
-                val oldFilter = filters[itemIndex]
-                val newFilter = Filter(
-                    oldFilter.id, binding.phraseEditText.text.toString(), oldFilter.context,
-                    oldFilter.expiresAt, oldFilter.irreversible, binding.phraseWholeWord.isChecked
-                )
-                updateFilter(newFilter, itemIndex)
-            }
-            .setNegativeButton(R.string.filter_dialog_remove_button) { _, _ ->
-                deleteFilter(itemIndex)
-            }
-            .setNeutralButton(android.R.string.cancel, null)
-            .show()
+            )
+        }
     }
 
     private fun refreshFilterDisplay() {
-        binding.filtersView.adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, filters.map { filter -> filter.phrase })
-        binding.filtersView.onItemClickListener = AdapterView.OnItemClickListener { _, _, position, _ -> setupEditDialogForItem(position) }
+        binding.filtersView.adapter = ArrayAdapter(
+            this,
+            android.R.layout.simple_list_item_1,
+            filters.map { filter ->
+                if (filter.expiresAt == null) {
+                    filter.phrase
+                } else {
+                    getString(
+                        R.string.filter_expiration_format,
+                        filter.phrase,
+                        DateUtils.getRelativeTimeSpanString(
+                            filter.expiresAt.time,
+                            Date().time,
```

Comment:
```suggestion
                            System.currentTimeMillis(),
```

(saves one object allocation)